lib/mogrify.ex

```diff
@@ -275,11 +276,38 @@ defmodule Mogrify do
     image_operator(image, "xc:#{color}")
   end
 
+  def add_option(image, option) do
+    validate_option!(option)
+    custom(image, option.name, option.argument)
+  end
+
   def custom(image, action, options \\ nil) do
     %{image | operations: image.operations ++ [{action, options}]}
   end
 
   def image_operator(image, operator) do
     %{image | operations: image.operations ++ [{:image_operator, operator}]}
   end
+
+  defp valid_option?(%Option{require_arg: true, argument: nil}), do: false
+  defp valid_option?(_), do: true
+
+  defp validate_option!(%Option{name: name} = option) do
+    case valid_option?(option) do
```

Comment:
If there's only true/false, why do you need a `case` here? if/else should be enough I'd say

Reply:
Good point, probably I tend to use case statements more often.